src/tests/dotnet-counters/CounterMonitorPayloadTests.cs

```diff
@@ -0,0 +1,338 @@
+
+                await TestRunnerUtilities.ExecuteCollection((ct) => {
+                    return Task.Run(async () =>
+                        await monitor.Collect(
+                            ct: ct,
+                            counter_list: counterList,
+                            counters: null,
+                            console: new TestConsole(),
+                            processId: testRunner.Pid,
+                            refreshInterval: 1,
+                            format: exportFormat,
+                            output: path,
+                            name: null,
+                            diagnosticPort: null,
+                            resumeRuntime: false,
+                            maxHistograms: 10,
+                            maxTimeSeries: 1000,
+                            duration: TimeSpan.FromSeconds(10)));
+                }, testRunner, source.Token);
+
+                return CreateMetricComponents();
+            }
+            finally
+            {
+                try
+                {
+                    File.Delete(path);
+                }
+                catch { }
+            }
+        }
+
+        private void ValidateCustomMetrics(List<MetricComponents> metricComponents, CountersExportFormat format)
+        {
+            // Currently not validating timestamp due to https://github.com/dotnet/diagnostics/issues/3905
+
+            HashSet<string> expectedProviders = new() { Constants.TestMeterName };
+            Assert.Equal(expectedProviders, metricComponents.Select(c => c.ProviderName).ToHashSet());
+
+            HashSet<string> expectedCounterNames = new() { Constants.TestHistogramName, Constants.TestCounterName };
+            Assert.Equal(expectedCounterNames, metricComponents.Select(c => c.CounterName).ToHashSet());
+
+            Assert.Equal(ExpectedCounterTypes, metricComponents.Select(c => c.CounterType).ToHashSet());
+
+            string tagSeparator = format == CountersExportFormat.csv ? ";" : ",";
+            string tag = Constants.TagKey + "=" + Constants.TagValue + tagSeparator + Constants.PercentileKey + "=";
+            HashSet<string> expectedTags = new() { $"{tag}{Constants.Quantile50}", $"{tag}{Constants.Quantile95}", $"{tag}{Constants.Quantile99}" };
+            Assert.Equal(expectedTags, metricComponents.Where(c => c.CounterName == Constants.TestHistogramName).Select(c => c.Tags).Distinct());
+            Assert.Empty(metricComponents.Where(c => c.CounterName == Constants.TestCounterName).Where(c => c.Tags != string.Empty));
+
+            var actualCounterValues = metricComponents.Where(c => c.CounterName == Constants.TestCounterName).Select(c => c.Value);
+
+            // We only expect 1, but due to timing variance some intervals may not have gotten a value (and others may have gotten two).
+            HashSet<double> expectedCounterValues = new() { 0, 1, 2 };
```

Comment:
I don't think we should be testing the specific value anymore. Let's just test that the value exists.